src/types/addons/mod.rs

```diff
@@ -9,6 +9,13 @@ use derive_more::*;
 
 pub type TransportUrl = String;
 
+#[derive(PartialEq, Clone, Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DescriptorPreview {
+    pub manifest: ManifestPreview,
```

Comment:
it will be useful in the future to replace that with some light manifest, which only contains name/description/images, and not the full catalogs info, which may get heavy